_deploy/p/gnoswap/pool/swap_math_test.gno

```diff
@@ -0,0 +1,206 @@
+			chkSqrtNextX96: func(sqrtRatioNextX96, priceTarget *u256.Uint) {
+				uassert.True(t, sqrtRatioNextX96.Eq(priceTarget))
+			},
+			amountIn:  "0",
+			amountOut: "0",
+			feeAmount: "10",
+		},
+		{
+			name:            "handles intermediate insufficient liquidity in zero for one exact output case",
+			currentX96:      u256.MustFromDecimal("20282409603651670423947251286016"),
+			targetX96:       u256.MustFromDecimal("22310650564016837466341976414617"),
+			liquidity:       u256.MustFromDecimal("1024"),
+			amountRemaining: i256.MustFromDecimal("-4"),
+			feePips:         3000,
+			sqrtNextX96:     u256.MustFromDecimal("22310650564016837466341976414617"),
+			chkSqrtNextX96: func(sqrtRatioNextX96, priceTarget *u256.Uint) {
+				uassert.True(t, sqrtRatioNextX96.Eq(priceTarget))
+			},
+			amountIn:  "26215",
+			amountOut: "0",
+			feeAmount: "79",
+		},
+		{
+			name:            "handles intermediate insufficient liquidity in one for zero exact output case",
+			currentX96:      u256.MustFromDecimal("20282409603651670423947251286016"),
+			targetX96:       u256.MustFromDecimal("18254168643286503381552526157414"),
+			liquidity:       u256.MustFromDecimal("1024"),
+			amountRemaining: i256.MustFromDecimal("-263000"),
+			feePips:         3000,
+			sqrtNextX96:     u256.MustFromDecimal("18254168643286503381552526157414"),
+			chkSqrtNextX96: func(sqrtRatioNextX96, priceTarget *u256.Uint) {
+				uassert.True(t, sqrtRatioNextX96.Eq(priceTarget))
+			},
+			amountIn:  "1",
+			amountOut: "26214",
+			feeAmount: "1",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			sqrtRatioNextX96, amountIn, amountOut, feeAmount := SwapMathComputeSwapStepStr(test.currentX96, test.targetX96, test.liquidity, test.amountRemaining, test.feePips)
+			test.chkSqrtNextX96(u256.MustFromDecimal(sqrtRatioNextX96), test.sqrtNextX96)
+			uassert.Equal(t, amountIn, test.amountIn)
+			uassert.Equal(t, amountOut, test.amountOut)
+			uassert.Equal(t, feeAmount, test.feeAmount)
+		})
+	}
+}
+
+// encodePriceSqrt calculates the sqrt((reserve1 << 192) / reserve0)
+func encodePriceSqrt(reserve1, reserve0 string) *u256.Uint {
+	reserve1Uint := u256.MustFromDecimal(reserve1)
+	reserve0Uint := u256.MustFromDecimal(reserve0)
+
+	if reserve0Uint.IsZero() {
+		panic("division by zero")
+	}
+
+	// numerator = reserve1 * (2^192)
+	two192 := new(u256.Uint).Lsh(u256.NewUint(1), 192)
+	numerator := new(u256.Uint).Mul(reserve1Uint, two192)
+
+	// ratioX192 = numerator / reserve0
+	ratioX192 := new(u256.Uint).Div(numerator, reserve0Uint)
+
+	// Return sqrt(ratioX192)
+	return sqrt(ratioX192)
+}
+
+// sqrt computes the integer square root of a u256.Uint
+func sqrt(x *u256.Uint) *u256.Uint {
+	if x.IsZero() {
+		return u256.NewUint(0)
+	}
+
+	z := new(u256.Uint).Set(x)
+	y := new(u256.Uint).Rsh(z, 1) // Initial guess is x / 2
+
+	for y.Cmp(z) < 0 {
+		z.Set(y)
+		temp := new(u256.Uint).Div(x, z)
+		y.Add(z, temp).Rsh(y, 1)
+	}
+	return z
+}
```

Comment:
It seems like these two functions are not using as test helper function

Reply:
https://github.com/gnoswap-labs/gnoswap/pull/429/commits/099f1cb15ad3d0863e728d01735d69aabe1471c3